pom.xml

```diff
@@ -19,7 +19,7 @@
   <parent>
     <groupId>org.springframework.boot</groupId>
     <artifactId>spring-boot-starter-parent</artifactId>
-    <version>1.2.7.RELEASE</version>
+    <version>1.4.7.RELEASE</version>
```

Comment:
figured i might use this opportunity to also upgrade spring a bit (higher versions give some problems with the mocking class atm)

Reply:
I am a little curious about why we need to explicitly disable the start up web context suddenly if it wasn't necessary for this project before. 

I suspect it's related to the updated spring boot version?

The `cmo-pipelines` is using `1.2.7` still so if the version updates here then I suspect there'll be some conflicts to resolve in `cmo-pipelines` as well. Is this upgrade required here or can it wiat?

Reply:
Ah I see - good point. I'll see if i can split out the spring upgrade

Reply:
@ao508 so yeah looks like that doesn't work b/c somehow the log configuration then breaks:

`Caused by: java.lang.NoClassDefFoundError: org/springframework/util/Log4jConfigurer`

I guess by upgrading spring boot it is able to use log4j from that one 🤔

Reply:
see e.g. the dep tree:
```
[INFO] +- org.springframework.boot:spring-boot-starter-batch:jar:1.4.7.RELEASE:compile
[INFO] |  +- org.springframework.boot:spring-boot-starter:jar:1.4.7.RELEASE:compile
[INFO] |  |  +- org.springframework.boot:spring-boot:jar:1.4.7.RELEASE:compile
[INFO] |  |  +- org.springframework.boot:spring-boot-autoconfigure:jar:1.4.7.RELEASE:compile
[INFO] |  |  +- org.springframework.boot:spring-boot-starter-logging:jar:1.4.7.RELEASE:compile
[INFO] |  |  |  +- org.slf4j:jcl-over-slf4j:jar:1.7.30:compile
[INFO] |  |  |  \- org.slf4j:jul-to-slf4j:jar:1.7.30:compile
```

Reply:
in `cmo-pipelines` we are also excluding  some logging dependencies from the spring boot starter batch dep


```
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-batch</artifactId>
      <exclusions>
        <exclusion>
          <groupId>org.slf4j</groupId>
          <artifactId>log4j-over-slf4j</artifactId>
        </exclusion>
        <exclusion>
          <groupId>ch.qos.logback</groupId>
          <artifactId>logback-classic</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
```

does it end up working if we do the same for this project while keeping the version 1.2.7?

Reply:
@ao508 that's actually already excluded here: 

https://github.com/genome-nexus/genome-nexus-annotation-pipeline/blob/09b0a0522d04b7b2bffa94c3b8c7df7f67da4a5f/pom.xml#L36-L46

> I suspect it's related to the updated spring boot version?

It's actually not related to spring boot version update in this PR turns out but rather to the dependency management changes in this PR: https://github.com/cBioPortal/cbioportal/pull/7984